### ISSUE TITLE
Default command invoked when calling .showHelp

### ISSUE THREAD

**Issue by wesleytodd** [yargs/yargs#1791]
```javascript
const yargs = require('yargs')

yargs.command('$0', 'a default command', (yargs) => yargs, () => {
  console.log('default command!! why?')
})
yargs.showHelp((output) => {
  console.log(output)
})
```

The above application will also execute the default command even though we never called `.argv` or `.parse`.  This seems like unexpected behavior to me and I could not find any docs on this.  Is this intended for some reason I am unaware of?

**Comment by bcoe** [comment]
@wesleytodd this definitely doesn't seem like intended behavior, thanks for the bug report.

**Comment by bcoe** [comment]
@wesleytodd I believe I fixed this as part of my work in #1823

There's also now a helper `await yargs.getHelp()`, which you could use rather than the callback based approach. I'd love feedback, you can install via:

`npm i yargs@next`
